app/scripts/lib/local-store.js

```diff
@@ -11,6 +12,38 @@ export default class ExtensionStore {
     if (!this.isSupported) {
       log.error('Storage local API not available.');
     }
+    this.dataPersistenceFailing = false;
+  }
+
+  setMetaData(initMetaData) {
+    this.metadata = initMetaData;
+  }
+
+  async persistStateToLocalStore(state) {
```

Comment:
Nit: This function could be simplified if it accepted the controller state directly. In practice, the `meta` property is never passed in here.

Reply:
Done [here](https://github.com/MetaMask/metamask-extension/pull/15978/commits/6d82b9026a036ba1b960d4e9659e73f542896dc7)